packages/edit-site/src/components/sidebar-navigation-screen-navigation-menu/index.js

```diff
@@ -60,120 +49,19 @@ export default function SidebarNavigationScreenNavigationMenu() {
 				getEditedEntityRecord: getEditedEntityRecordSelector,
 			};
 		},
-		[ postId, postType ]
+		[ postId ]
 	);
 
 	const isLoading = isResolving || isSaving || isDeleting;
 
 	const menuTitle = navigationMenu?.title?.rendered || navigationMenu?.slug;
 
-	const handleSave = async ( edits = {} ) => {
-		// Prepare for revert in case of error.
-		const originalRecord = getEditedEntityRecord(
-			'postType',
-			'wp_navigation',
-			postId
-		);
-
-		// Apply the edits.
-		editEntityRecord( 'postType', postType, postId, edits );
+	const { handleSave, handleDelete, handleDuplicate } =
+		useNavigationMenuHandlers();
 
-		// Attempt to persist.
-		try {
-			await saveEditedEntityRecord( 'postType', postType, postId, {
-				throwOnError: true,
-			} );
-			createSuccessNotice( __( 'Renamed Navigation menu' ), {
-				type: 'snackbar',
-			} );
-		} catch ( error ) {
-			// Revert to original in case of error.
-			editEntityRecord( 'postType', postType, postId, originalRecord );
-
-			createErrorNotice(
-				sprintf(
-					/* translators: %s: error message describing why the navigation menu could not be renamed. */
-					__( `Unable to rename Navigation menu (%s).` ),
-					error?.message
-				),
-
-				{
-					type: 'snackbar',
-				}
-			);
-		}
-	};
-
-	const handleDelete = async () => {
-		try {
-			await deleteEntityRecord(
-				'postType',
-				postType,
-				postId,
-				{
-					force: true,
-				},
-				{
-					throwOnError: true,
-				}
-			);
-			createSuccessNotice( __( 'Deleted Navigation menu' ), {
-				type: 'snackbar',
-			} );
-			goTo( '/navigation' );
-		} catch ( error ) {
-			createErrorNotice(
-				sprintf(
-					/* translators: %s: error message describing why the navigation menu could not be deleted. */
-					__( `Unable to delete Navigation menu (%s).` ),
-					error?.message
-				),
-
-				{
-					type: 'snackbar',
-				}
-			);
-		}
-	};
-	const handleDuplicate = async () => {
-		try {
-			const savedRecord = await saveEntityRecord(
-				'postType',
-				postType,
-				{
-					title: sprintf(
-						/* translators: %s: Navigation menu title */
-						__( '%s (Copy)' ),
-						menuTitle
-					),
-					content: navigationMenu?.content?.raw,
-					status: 'publish',
-				},
-				{
-					throwOnError: true,
-				}
-			);
-
-			if ( savedRecord ) {
-				createSuccessNotice( __( 'Duplicated Navigation menu' ), {
-					type: 'snackbar',
-				} );
-				goTo( `/navigation/${ postType }/${ savedRecord.id }` );
-			}
-		} catch ( error ) {
-			createErrorNotice(
-				sprintf(
-					/* translators: %s: error message describing why the navigation menu could not be deleted. */
-					__( `Unable to duplicate Navigation menu (%s).` ),
-					error?.message
-				),
-
-				{
-					type: 'snackbar',
-				}
-			);
-		}
-	};
+	const _handleDelete = () => handleDelete( navigationMenu );
```

Comment:
Is this a pattern we use elsewhere? Another option could be to call it `handleDeleteBound`...

Reply:
Yeh I mean...it isn't exactly elegant 🤷 

I'm not 100% about the API on this hook yet. Let's allow new _requirements_ to drive any further refactoring however.